### PR TITLE
[SPARK-11799][CORE] Make it explicit in executor logs that uncaught e…

### PR DESCRIPTION
**Pull request by vundela**
…xceptions are thrown during executor shutdown

This commit will make sure that when uncaught exceptions are prepended with [Container in shutdown] when JVM is shutting down.

### CODE REVIEW

**Comment by vanzin** [comment]
ok to test

File: core/src/main/scala/org/apache/spark/util/SparkUncaughtExceptionHandler.scala [hunk 1]
```diff
@@ -46,6 +46,18 @@ private[spark] object SparkUncaughtExceptionHandler
     }
   }
 
+  def log(thread: Thread, exception: Throwable): Unit = {
+    val errMsg = "Uncaught exception in thread "
+    val inShutdownMsg = "[Container in shutdown] "
```
Comment (vanzin) [inline core/src/main/scala/org/apache/spark/util/SparkUncaughtExceptionHandler.scala]: minor: less code if done like this:

    val inShutdownMessage = if (ShutdownHookManager.inShutdown()) "[Container in shutdown] " else ""
    logError(inShutdownMsg + errMsg + thread, exception)

File: core/src/main/scala/org/apache/spark/util/SparkUncaughtExceptionHandler.scala [hunk 2]
```diff
@@ -46,6 +46,15 @@ private[spark] object SparkUncaughtExceptionHandler
     }
   }
 
+  def log(thread: Thread, exception: Throwable): Unit = {
+    val errMsg = "Uncaught exception in thread "
+
+    // Make it explicit that uncaught exceptions are thrown when container is shutting down.
+    // It will help users when they analyze the executor logs
+    val inShutdownMsg = if (ShutdownHookManager.inShutdown()) "[Container in shutdown] " else ""
+    logError(inShutdownMsg + errMsg + thread, exception)
```
Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/util/SparkUncaughtExceptionHandler.scala]: this doesn't have to be its own method. I would just inline it

**Comment by andrewor14** [comment]
Merging into master 1.6.

**Pull request closed without merging**